src/main/io/osd.c

```diff
@@ -673,34 +673,41 @@ static bool osdDrawSingleElement(uint8_t item)
 
 #ifdef USE_ESC_SENSOR
             // Show warning if we lose motor output or the ESC is overheating
-            if (feature(FEATURE_ESC_SENSOR) && enabledWarnings & OSD_WARNING_ESC_FAIL) {
-                char escErrMsg[OSD_FORMAT_MESSAGE_BUFFER_SIZE];
+            if (feature(FEATURE_ESC_SENSOR) && enabledWarnings & OSD_WARNING_ESC_FAIL && ARMING_FLAG(ARMED)) {
```

Comment:
Why did you move the arming check to here? ESCs can show a temperature alarm even when unarmed.

Reply:
You're right. I noticed it wasnt being displayed when disarmed, but I've re-checked and it was only because another warning message had taken priority. (Maybe in the future need to support alternating the display between multiple messages).

Reply:
The idea here is to always display the warning with the highest priority, and the warnings are ordered in order of priority. This should probably move up to right underneath `BATTERY_CRITICAL`.